api/src/main/java/org/datacleaner/api/RestrictedFunctionalityCallToAction.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.api;
+
+import java.io.Serializable;
+
+/**
+ * Represents a "Call to Action" for users when restricted functionality is
+ * blocked, typically with the use of {@link RestrictedFunctionalityException}
+ * or {@link RestrictedFunctionalityMessage}.
+ */
+public class RestrictedFunctionalityCallToAction implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private final String _name;
```

Comment:
Right now  this is a string, but I think we could change it to an enum. The reasoning being that if we catch this in the UI, we probably want to show an advert specific towards either a buy credits page or a upgrade to enterprise page. Right now we would have to do with a generic popup and a URL. If the goal is to motivate users into actions, I think we need something more specific.

The other thing is that the URL right now has to be coded in the client, so if this url ever changes, we have to go through every transformer/analyzer and change it to the new one.

The description I would keep, so that the component can specify the details a bit more, eg. ""name cleansing requires x credits", "personal edition of duplicate detection is limited to x records".